### PR TITLE
CMake: Simplify run_test.sh and avoid array handling magic

### PR DESCRIPTION
**Pull request by tamiko**
In reference to #3775

### CODE REVIEW

File: cmake/scripts/run_test.sh [hunk 1]
```diff
@@ -18,143 +18,146 @@
...
-NUMDIFF_EXECUTABLE="$3"
-DIFF_EXECUTABLE="$4"
-COMPARISON_FILE="$5"
-shift 5
-
-save () {
-  for i do printf %s\\n "$i" | sed "s/'/'\\\\''/g;1s/^/'/;\$s/\$/' \\\\/" ;
-  done
-  echo " "
-}
```
Comment (tamiko) [inline cmake/scripts/run_test.sh]: @tjhei Simply restructuring the shell script and removing this arcane magic is also an option.

File: cmake/scripts/run_test.sh [hunk 2]
```diff
@@ -18,143 +18,146 @@
...
-    # determine variant name (empty string for main comparison file):
-    variant="${file#*.output}"
+    test_successful=false
 
     #
-    # Run diff or numdiff (if available) to determine whether files are the
-    # same. Create a diff file "diff${variant}" for each variant file that
-    # is found (including the main comparison file).
+    # Pick up main comparison file and all variants. A valid variant name is
+    # of the form [...].output.[STRING]
```
Comment (tjhei) [inline cmake/scripts/run_test.sh]: You don't enforce the "." after output though, right?

Comment (tamiko) [inline cmake/scripts/run_test.sh]: No, I probably don't enforce it. Judging from the snippet below, we pick up everything that matches `"${COMPARISON_FILE}"*`.

Comment (tamiko) [inline cmake/scripts/run_test.sh]: Now, I also remember the reason. The old variant was
```
for files in "${COMPARISON_FILE}" "${COMPARISON_FILE}".*; ...
```
This produces at least a warning if the second expression doesn't match anything.

Comment (tjhei) [inline cmake/scripts/run_test.sh]: fair enough.

**Pull request merged**